ForumSessionProvider/extension.json

```diff
@@ -37,7 +45,8 @@
     ]
   },
   "AutoloadClasses": {
-    "ForumSessionProvider": "ForumSessionProvider.php"
+    "ForumSessionProvider": "ForumSessionProvider.php",
+    "AuthManager_SMF": "ForumSessionProvider.php"

```

Comment:
If you want to keep this a separate classes, can just as well add a second file under the name.

Also worry that SMF branding may confuse people in thinking it's more specific than it is.